rllib/algorithms/impala/impala_learner.py

```diff
@@ -50,25 +54,35 @@ class ImpalaLearner(Learner):
     def build(self) -> None:
         super().build()
 
-        # Build entropy coeff scheduling tools.
-        self.entropy_coeff_scheduler = Scheduler(
-            fixed_value=self.hps.entropy_coeff,
-            schedule=self.hps.entropy_coeff_schedule,
-            framework=self.framework,
-            device=self._device,
+        # Dict mapping module IDs to the respective entropy Scheduler instance.
+        self.entropy_coeff_schedulers_per_module: Dict[
+            ModuleID, Scheduler
+        ] = LambdaDefaultDict(
+            lambda module_id: Scheduler(
+                fixed_value_or_schedule=(
+                    self.hps.get_hps_for_module(module_id).entropy_coeff
+                ),
+                framework=self.framework,
+                device=self._device,
+            )
         )
 
+    @override(Learner)
+    def remove_module(self, module_id: str):
+        super().remove_module(module_id)
+        self.entropy_coeff_schedulers_per_module.pop(module_id)
+
     @override(Learner)
     def additional_update_for_module(
-        self, *, module_id: ModuleID, timestep: int
+        self, *, module_id: ModuleID, hps: ImpalaLearnerHyperparameters, timestep: int
     ) -> Dict[str, Any]:
         results = super().additional_update_for_module(
-            module_id=module_id, timestep=timestep
+            module_id=module_id, hps=hps, timestep=timestep
         )
 
         # Update entropy coefficient via our Scheduler.
-        new_entropy_coeff = self.entropy_coeff_scheduler.update(
-            module_id, timestep=timestep
+        new_entropy_coeff = self.entropy_coeff_schedulers_per_module[module_id].update(
```

Comment:
I think this update should be tied to a input flag so that the user can control the frequency of each additional update event, independently from other kind of updates.

Reply:
good idea, will add

Reply:
Hmm, this doesn't make sense, imo. Then we should simply add different methods, one for each type of update, e.g. `update_entropy_coeff`, just like we had it in the old `PPOPolicies.update_kl()`.

I think that the PPO algorithm has to trust its framework-specific Learner subclasses to do the right thing (PPO-wise) in this one `additional_update` call.

Also, in this particular case (updating the entropy coeff), it really doesn't make sense to have more control. There is no reason to NOT update this coeff every single iteration.

I do see that this situation is different for say target net updates, where you don't want these updates to happen every iteration. So for these, we should indeed have a flag (rather than deciding inside additional_update by the `timestep` given as the algorithm should be the one deciding WHEN to do WHAT).

Bottom line: Yes, for target net updates, this makes sense. For lr/entropy/kl coeff updates not so much.